pennylane/templates/layers/gate_fabric.py

```diff
@@ -0,0 +1,235 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the QuantumNumberPreserving Gate Fabric template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class GateFabric(Operation):
+    r"""Implements a local, expressive, and quantum-number-preserving ansatz using VQE circuit fabrics
+    proposed by Anselmetti *et al.* in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05695>`_.
+
+    This template prepares the :math:`N` qubits trial state by applying :math:`L` layers of the gate fabric, which is composed
+    of 2-parameter 4-qubit gate elements :math:`\hat{Q}(\theta, \varphi)`. Each of the :math:`\hat{Q}(\theta, \varphi)` gate is itself
+    composed of two 1-parameter 4-qubit gates. The first gate is a spin-adapted spatial orbital rotation gate, which is implemented by
+    :math:`\text{OrbitalRotation}(\varphi)` gate. Whereas the second gate is a diagonal pair-exchange gate, which is implemented by the
+    :math:`\text{DoubleExcitation}(\theta)` gate. In addition to these two gates, :math:`\hat{Q}(\theta, \varphi)` can also include an
+    optional gate :math:`\hat{\Pi} \in \{\hat{I}, \text{OrbitalRotation}(\pi)\}` gate, whose non-identity value could be advantageous in
+    gradient-based parameter optimization. Regardless, of the choice of :math:`\hat{\Pi}`, this gate fabric will exactly preserve the
+    quantum numbers :math:`\hat{N}_{\alpha}`, :math:`\hat{N}_{\beta}` and :math:`\hat{S}^{2}`.
+
+    The circuit implementing the gate fabric layer for `N = 8` is shown below:
+
+    |
+
+    .. figure:: ../../_static/templates/layers/gate_fabric_layer.png
+        :align: center
+        :width: 40%
+        :target: javascript:void(0);
+
+    |
+
+    The 2-parameter, 4-qubit :math:`\hat{Q}(\theta, \varphi)` gate is decomposed as follows:
+
+    |
+
+    .. figure:: ../../_static/templates/layers/q_gate_decompositon.png
+        :align: center
+        :width: 100%
+        :target: javascript:void(0);
+
+    |
+
+    The individual :math:`\text{DoubleExcitation}(\theta)` and  :math:`\text{OrbitalRotation}(\varphi)` gates are decomposed using
+    Givens rotation gate :math:`G(\phi)`, which is implemented in pennylane with :math:`\text{SingleExcitation}(phi)`.
+
+    |
+
+    .. figure:: ../../_static/templates/layers/double_excitation_decomposition.png
+        :align: center
+        :width: 80%
+        :target: javascript:void(0);
+
+    .. figure:: ../../_static/templates/layers/orbital_rotation_decomposition.png
+        :align: center
+        :width: 80%
+        :target: javascript:void(0);
+
+    .. figure:: ../../_static/templates/layers/givens_rotation_decomposition.png
+        :align: center
+        :width: 100%
+        :target: javascript:void(0);
+
+    |
```

Comment:
Actually, now that I look at the other templates, I think it is better to not have the decomposed gates here. They can appear in their own page. Maybe just mention specifically that the QNP_PX gate in the original ref is equivalent to Double excitation (are we sure that the CNOT gates in QNP_PX can be neglected?).

Reply:
I have removed the decompositions, and instead, just mentioned that the gates `QNP_OR` and `QNP_PX` given in the original references are equivalent to the `OrbitalRotation` and `DoubleExcitation` gates.